onnxscript/irbuilder.py

```diff
@@ -108,10 +108,16 @@ def _opt_var_to_str(x):
 
 
 class IRAttributeValue:
-    """An attribute value (representing an actual parameter)."""
+    """An attribute value (representing an actual parameter).
```

Comment:
With the current changes, this class is being used to cover both *actual* parameters (which are values) as well as *formal* parameters (with or without a default-value). I am not 100% sure if we should unify all of these into a single type/class. But if we do, may be the name should be changed to reflect this? May be `IRAttributeParameter` ? And, of course, the above documentation line should be changed to actual+formal parameter.

Reply:
For example, should we add an assertion/check in the IRStmt __init__ method that all attrs have a default-value?

Reply:
Thanks! I added IRAttributeParameter. I wonder if it should be a subclass of IRAttributeValue or be its own class? For now I subclassed but semantically should it be that?

Reply:
Yes, semantically it should not be a subclass. The current form is preferable.